examples/vision/fully_convolutional_network.py

```diff
@@ -0,0 +1,651 @@
+In this example, we are going to work on developing a Fully-Convolutional Network that is
+capable of performing Image Segmentation. For this, we make use of the VGG-19 model as
+available in `keras.applications` and continue towards extending its internal layers and
+creating our Fully-Convolutional Network.
+The network extends the pooling layer outputs from the VGG in order to perform upsampling
+and get a final result. Different intermediate pool layers are extracted and processed
+upon for different versions of the network.
+The FCN architecture has 3 versions of differing quality, as mentioned in the paper.
+- FCN-32S
+- FCN-16S
+- FCN-8S
+All versions of the model derive their outputs through an iterative processing of
+successive intermediate pool layers of the main backbone used. A better idea can be
+gained from the figure below.
+| ![FCN Architecture](https://i.imgur.com/Ttros06.png) |
+| :--: |
+| **Diagram 1**: Combined Architecture Versions (Source: Paper)|
+
+The process we are going to follow is:
+- Set-up all imports
+- Set global notebook configuration
+- Load and preprocess dataset
+- Visualize random samples from pre-processed dataset
+- Define our model in steps
+  - Backbone (VGG-19)
+  - FCN-32S
+  - FCN-16S
+  - FCN-8S
+- Load weights into converted Convolution layers
+- Training
+  - FCN-32S
+  - FCN-16S
+  - FCN-8S
+- Visualizations
+  - Metrics
+  - Segmentation Mask results
+
+To get a better idea on Image Segmentation or find more pre-trained models, feel free to
+navigate to the links below
+- [Hugging Face Image
+Segmentation](https://huggingface.co/models?pipeline_tag=image-segmentation)
+- [PyImageSearch Blog on Semantic
+Segmentation](https://pyimagesearch.com/2018/09/03/semantic-segmentation-with-opencv-and-deep-learning/)
+
+"""
+
+"""
+## Setup Imports
+
+We use the imports for the following reasons
+- `tensorflow_datasets`: Utility to download our dataset
+- `matplotlib.pyplot`: Utility for plotting and visualizations
+- `numpy`: Utility for performing array manipulation operations
+
+We then set the Random Seed to make sure the results are reproducible over separate
+iterations. We also use `tf.data.AUTOTUNE` to set some parameters based on the current
+system's configuration.
+"""
+
+import tensorflow as tf
+from tensorflow import keras
+import matplotlib.pyplot as plt
+import tensorflow_datasets as tfds
+import numpy as np
+
+keras.utils.set_random_seed(42)
+
+AUTOTUNE = tf.data.AUTOTUNE
+
+"""
+## Set configurations for notebook variables
+
+We set the required parameters for the experiment. The chosen dataset has a total of 4
+classes per image, with regards to the segmentation mask. We also set our hyperparameters
+in this cell.
+
+Mixed Precision as an option is also available in systems which support it, to reduce
+load. This would make most tensors use `16-bit float` values instead of `32-bit float`
+values, in places where it will not adversely affect computation.
+"""
+
+config = {
```

Comment:
Prefer variables to a config dictionary!  

Doing this in a dictionary doesn't provide any value over global variables.